src/compiler/emitter.ts

```diff
@@ -3660,13 +3661,13 @@ namespace ts {
 
         function hasTrailingCommentsAtPosition(pos: number) {
             let result = false;
-            forEachTrailingCommentRange(currentSourceFile?.text || "", pos + 1, () => result = true);

```

Comment:
@sandersn I did this because I wanted to avoid `JSDoc` nodes duplication in `.d.ts` file, e.g.
```
// file.js
/**
 * @overload
 * @param {string} value
 * @return {string}
 */
/**
 * @overload
 * @param {number} value
 * @return {number}
 */
/**
 * @overload
 * @param {unknown} value
 */
function myFunction(value) {}
```
would result in:
```
// file.d.ts
/**
 * @overload
 * @param {string} value
 */
declare function myFunction(value: string): string;
/**
 * @overload
 * @param {number} value
 */
declare function myFunction(value: number): number;
/**
 * @overload
 * @param {string} value
 */
/**
 * @overload
 * @param {number} value
 */
/**
 * @overload
 * @param {unknown} value
 */
declare function myFunction(value);
```
Being able to specify `end` allowed me to limit the `JSDoc`s that are emitted for a given parent node.

However this seems to only be one of many weird consequences of synthesizing declarations, which as you suggested does not seem to be the proper approach anyway.